pkg/theiactl/commands/start.go

```diff
@@ -0,0 +1,378 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strconv"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	sparkv1 "antrea.io/theia/third_party/sparkoperator/v1beta2"
+)
+
+const (
+	flowVisibilityNS     = "flow-visibility"
+	k8sQuantitiesReg     = "^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$"
+	sparkImage           = "antrea/theia-policy-recommendation:latest"
+	sparkImagePullPolicy = "IfNotPresent"
+	sparkAppFile         = "local:///opt/spark/work-dir/policy_recommendation_job.py"
+	sparkServiceAccount  = "policy-reco-spark"
+	sparkVersion         = "3.1.1"
+)
+
+type SparkResourceArgs struct {
+	executorInstances   int32
+	driverCoreRequest   string
+	driverMemory        string
+	executorCoreRequest string
+	executorMemory      string
+}
+
+// startCmd represents the start command
+var startCmd = &cobra.Command{
+	Use:   "start",
+	Short: "Start a new policy recommendation Spark job",
+	Long: `Start a new policy recommendation Spark job.
+Network policies will be recommended based on the flow records sent by Flow Aggregator.
+Must finish the deployment of Theia first, please follow the steps in 
+https://github.com/antrea-io/theia/blob/main/docs/network-policy-recommendation.md`,
+	Example: `Start a policy recommendation spark job with default configuration
+$ theiactl policyreco start
+Start an initial policy recommendation spark job with network isolation option 1 and limit on last 10k flow records
+$ theiactl policyreco start --type initial --option 1 --limit 10000
+Start an initial policy recommendation spark job with network isolation option 1 and limit on flow records from 2022-01-01 00:00:00 to 2022-01-31 23:59:59.
+$ theiactl policyreco start --type initial --option 1 --start_time '2022-01-01 00:00:00' --end_time '2022-01-31 23:59:59'
+Start a policy recommendation spark job with default configuration but doesn't recommend toServices ANPs
+$ theiactl policyreco start --to_services=false
+`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		var recoJobArgs []string
+		sparkResourceArgs := SparkResourceArgs{}
+
+		recoType, err := cmd.Flags().GetString("type")
+		if err != nil {
+			return err
+		}
+		if recoType != "initial" && recoType != "subsequent" {
+			return fmt.Errorf("recommendation type should be 'initial' or 'subsequent'")
+		}
+		recoJobArgs = append(recoJobArgs, "--type", recoType)
+
+		limit, err := cmd.Flags().GetInt("limit")
+		if err != nil {
+			return err
+		}
+		if limit < 0 {
+			return fmt.Errorf("limit should be an integer >= 0")
+		}
+		recoJobArgs = append(recoJobArgs, "--limit", strconv.Itoa(limit))
+
+		option, err := cmd.Flags().GetInt("option")
+		if err != nil {
+			return err
+		}
+		if option < 1 || option > 3 {
```

Comment:
I would prefer to map these integer values to options that ore more intelligible.

For instance instead of --option 3 can user specify --use-k8s-net-pol?

Reply:
Thanks!